server/jetstream_api.go

```diff
@@ -1825,17 +1829,37 @@ func (s *Server) jsStreamInfoRequest(sub *subscription, c *client, a *Account, s
 	// Check if they have asked for subject details.
 	if subjects != _EMPTY_ {
 		if mss := mset.store.SubjectsState(subjects); len(mss) > 0 {
-			if len(mss) > JSMaxSubjectDetails {
-				resp.StreamInfo = nil
-				resp.Error = NewJSStreamInfoMaxSubjectsError()
-				s.sendAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
-				return
+			// As go iterates over map in a non-consistent order, no choice but to buffer it a slice
+			type SubjectsState struct {
+				subject  string
+				messages uint64
 			}
-			sd := make(map[string]uint64, len(mss))
+
+			buffer := make([]SubjectsState, 0, len(mss))
```

Comment:
Why not simply a slice of strings since we care only about sorting the subjects, not the combination of subject and messages. At the end, when it is time to "feed" the map `sd` you would do something like `sd[subject] = mss[subject]`.

(because right now it is worse that you not make it a []*SubjectsState because sort is then done moving structures in that slice, as opposed to move pointers).